distributed/comm/ucx.py

```diff
@@ -175,7 +171,8 @@ def peer_address(self) -> str:
                 #  syncing the default stream will wait for other non-blocking CUDA streams.
                 # Note this is only sufficient if the memory being sent is not currently in use on
                 # non-blocking CUDA streams.
-                synchronize_stream(0)
+                if any(cuda_frames):
```

Comment:
Then call the `.any()` method.

```suggestion
                if cuda_frames.any():
```